### DISCUSSION TITLE
Feature-Request: "Load multiple local tracks"

### DISCUSSION THREAD

**Discussion by Tomato6966** [lavalink-devs/lavaplayer#3] [Q&A]
## Description
loadtracks by providing a folder of mp3 files (I wouldn't allow nested requests just for the folder *.mp3|*.mov|*.ogg|etc.etc.

when doing `/loadtracks?identifier=/home/path/to/folder/` it loads all existing possible files which load the tracks
when doing `/loadtracks?identifier=/home/path/to/folder/*.mp3` it only loads files ending with .mp3 (aka glob support)

LoadType should be: "playlist"

Environmental Variable Suggestion:
localfilespages, each page allows to load up to 100 tracks

How I'd do it on nodejs manually, which isn't quite fast...
```js
import { readdir } from "fs/promises";
// makeRequest is for my lavalink server, which uses this lavaplayer repository.
async function loadFolderPlaylist(path) {
  const files = await readdir(path).then(files => files.filter(fileName => /(.mp3|.mov)/igu.test(fileName)));
  const tracks = await Promise.all(files.map(async file => await makeRequest(`/loadtracks?identifier=${path}/${file}`)));
  return Utils.BuildPlaylistResponse(tracks)
}
```

**Comment by topi314** [comment]
If you're gonna open a feature request you don't follow the bug report template

**Comment by freyacodes** [comment]
What do you hope to gain from having Lavalink index a directory instead of what you're already doing?

**Comment by Tomato6966** [comment]
It should be easier and quicker
Also wouldn't it be cool to support that?

**Reply by freyacodes** [reply]
You're already sending asynchronous requests by the looks of it. I don't see any speed gain here. In fact, unless we'd be careful to make it internally asynchronous it would actually be significantly slower.

>  Also wouldn't it be cool to support that?

Local track support is an afterthought to begin with

**Reply by Tomato6966** [reply]
hmm ok

**Reply by topi314** [reply]
The only advantage I can see here is that if your node is remote you usually don't have access to the filesystem, like this you can just load a folder without exactly knowing which files are in there
